### DISCUSSION TITLE
$watch callback triggered by unwatched data

### DISCUSSION THREAD

**Discussion by DiskoPete** [alpinejs/alpine#1627] [7. Bugs (archived)]
## Summary
I stumbled across an issue where an Alpine data triggers a `$watch` callback of another data.

To demonstrate the issue I created this codepen snippet:\
https://codepen.io/DiskoPete/pen/MWpdNpX

## Steps to reproduce
1. Change value B
2. Note that `b` is watched and will trigger `this.submit()`
3. Afterwards enter text for value a

## Results expected
`$watch` callback for value `b` isn't triggered when we change value `a`

## Actual results
After the value `b` `$watch` callback is triggered for the first time, the callback will also be fired if we change value `a`.

## OS/Browser
macOS 11.4; Chrome 91; alpinejs 3.1.0

## Other
Thank you very much for your time and effort you put into this! 🚀

**Comment by vladrusu** [comment]
I have a similar issue:
https://github.com/alpinejs/alpine/discussions/1545
